app/src/main/java/com/android/unio/ui/home/Home.kt

```diff
@@ -102,152 +90,199 @@ fun HomeScreen(
+                                x = progressFromFirstPage * tabWidth + tabWidth * 2 / 3,
+                                y = height - 25)
+
+                        drawLine(
+                            start = startOffset,
+                            end = endOffset,
+                            color = colorScheme.primary,
+                            strokeWidth = Stroke.DefaultMiter)
+                      })
+            }) {
+              val tabTestTags = listOf(HomeTestTags.TAB_ALL, HomeTestTags.TAB_FOLLOWING)
+              list.forEachIndexed { index, str ->
+                Tab(
+                    selected = index == pagerState.currentPage,
+                    // animate pager if click on tab
+                    onClick = { scope.launch { pagerState.animateScrollToPage(index) } },
+                    modifier =
+                        Modifier.testTag(tabTestTags[index]).onSizeChanged {
+                          sizeList[index] = Pair(it.width.toFloat(), it.height.toFloat())
+                        },
+                    selectedContentColor = colorScheme.primary) {
+                      Text(
+                          text = str,
+                          style =
+                              TextStyle(
+                                  fontWeight = FontWeight.Bold,
+                                  fontSize = 12.sp,
+                              ),
+                          modifier =
+                              Modifier.align(CenterHorizontally)
+                                  .padding(horizontal = 32.dp, vertical = 16.dp))
                     }
-                  }
+              }
             }
-          } else if (events.isNotEmpty()) {
-            LazyColumn(
-                contentPadding = PaddingValues(vertical = 8.dp),
-                modifier = Modifier.fillMaxSize().padding(horizontal = 32.dp)) {
-                  items(events) { event ->
-                    EventCard(navigationAction, event, userViewModel, eventViewModel)
-                  }
-                }
-          } else {
-            Box(
-                modifier = Modifier.fillMaxSize().padding(paddingValues),
-                contentAlignment = Alignment.Center) {
-                  Text(
-                      modifier = Modifier.testTag(HomeTestTags.EMPTY_EVENT_PROMPT),
-                      text = context.getString(R.string.event_no_events_available))
-                }
-          }
-        }
-      })
+
+        DockedSearchBar(
+            inputField = {
+              SearchBarDefaults.InputField(
+                  modifier = Modifier.testTag(HomeTestTags.SEARCH_BAR_INPUT),
+                  query = searchQuery,
+                  onQueryChange = {
+                    // Search when query changes
+                    searchQuery = it
+                    searchViewModel.debouncedSearch(it, SearchViewModel.SearchType.EVENT)
+                  },
+                  onSearch = {},
```

Comment:
Is it mandatory to define onSearch ? If not, it can be removed

Reply:
Yes, I just tried, onSearch is mandatory sadly